api/registrations/serializers.py

```diff
@@ -423,11 +435,17 @@ def _find_orphan_files(self, registering, draft):
 
 class RegistrationDetailSerializer(RegistrationSerializer):
     """
-    Overrides RegistrationSerializer to make id required.
+    Overrides RegistrationSerializer to make id required, and to make withdrawn and withdrawal_justification writable.
     """
 
     id = IDField(source='_id', required=True)
 
+    withdrawn = ser.BooleanField(
+        source='is_retracted', required=False,
+        help_text='The registration has been withdrawn.',
+    )
+    withdrawal_justification = ser.CharField(source='retraction.justification', required=False)
```

Comment:
One more thing, I don't think https://github.com/CenterForOpenScience/osf.io/pull/8836/ will break this, and the source of these fields should stay as-is.   PLAT-8836 changes the `withdrawal_justification` field source to point to a new property of the same name on the Registration model.  (This is so a registration's child will properly display the withdrawal justification).    But when we're writing to this field, the source should still be the justification field on the retraction.

Reply:
hmph, I was wrong, detail GET RegistrationDetail is now broken for children. fixing now!